my.css

```diff
@@ -16,10 +20,23 @@ body {
   outline: 2px solid green;
 }
 
-.paragraph {
-  font-size:  16px;
+// issue1: nesting
+body {
+  .product {
+    .product-imag {
+      .paragraph {
+        font-size:  1rem;
+      }
+    }
+  }
```

Comment:
A nifty set of problems coming from this innocent idea of organizing the content in neat blocks of css.

**Problem 1: (DOM mirroring)**
This is exactly replicating the DOM structure which makes the CSS very dependent of the markup
which will cause trouble in the future.

**Problem 2: (Too much nesting)**
This is nesting for no reason, the flat version would work too
which would also prevent the high cognitive load on the reader's mind
who constantly need to resolve the full and normalized selector

**Problem 3: (over-specificity)**
By using such a deep nesting it's also over specifying the target for no reason. A simple .header-link